File: system-test/java/src/test/java/com/optivem/shop/systemtest/latest/acceptance/PlaceOrderPositiveTest.java
package com.optivem.shop.systemtest.latest.acceptance;

import com.optivem.shop.systemtest.latest.acceptance.base.BaseAcceptanceTest;
import com.optivem.shop.dsl.channel.ChannelType;
import com.optivem.shop.dsl.driver.port.shop.dtos.OrderStatus;
import com.optivem.testing.Channel;

import org.junit.jupiter.api.TestTemplate;

class PlaceOrderPositiveTest extends BaseAcceptanceTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void orderNumberShouldStartWithORD() {
        scenario
                .when().placeOrder()
                .then().shouldSucceed()
                .and().order()
                    .hasOrderNumberPrefix("ORD-");
    }

    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void orderStatusShouldBePlacedAfterPlacingOrder() {
        scenario
                .when().placeOrder()
                .then().shouldSucceed()
                .and().order()
                    .hasStatus(OrderStatus.PLACED);
    }

    @TestTemplate
    @Channel(ChannelType.API)
    void orderTotalShouldIncludeTax() {
        scenario
                .when().placeOrder()
                    .withCountry("DE")
                .then().shouldSucceed()
                .and().order()
                    .hasSubtotalPrice(20.00)
                    .hasTaxRate(0.19)
                    .hasTotalPrice(23.80);
    }

    @TestTemplate
    @Channel(ChannelType.API)
    void orderTotalShouldReflectCouponDiscount() {
        scenario
                .given().coupon()
                    .withCode("DISC10")
                    .withDiscountRate(0.10)
                .when().placeOrder()
                    .withCouponCode("DISC10")
                .then().shouldSucceed()
                .and().order()
                    .hasSubtotalPrice(18.00)
                    .hasDiscountRate(0.10)
                    .hasAppliedCouponCode("DISC10")
                    .hasTotalPrice(18.00);
    }

    @TestTemplate
    @Channel(ChannelType.API)
    void orderTotalShouldApplyCouponDiscountAndTax() {
        scenario
                .given().coupon()
                    .withCode("COMBO10")
                    .withDiscountRate(0.10)
                .when().placeOrder()
                    .withCountry("GB")
                    .withCouponCode("COMBO10")
                .then().shouldSucceed()
                .and().order()
                    .hasSubtotalPrice(18.00)
                    .hasDiscountRate(0.10)
                    .hasTaxRate(0.20)
                    .hasAppliedCouponCode("COMBO10")
                    .hasTotalPrice(21.60);
    }
}

File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/PlaceOrderPositiveTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Driver.Port.Shop.Dtos;
using Optivem.Testing;

namespace SystemTests.Latest.AcceptanceTests;

public class PlaceOrderPositiveTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task OrderNumberShouldStartWithORD(Channel channel)
    {
        await Scenario(channel)
            .When().PlaceOrder()
            .Then().ShouldSucceed()
            .And().Order()
            .HasOrderNumberPrefix("ORD-");
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task OrderStatusShouldBePlacedAfterPlacingOrder(Channel channel)
    {
        await Scenario(channel)
            .When().PlaceOrder()
            .Then().ShouldSucceed()
            .And().Order()
            .HasStatus(OrderStatus.Placed);
    }

    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task OrderTotalShouldIncludeTax(Channel channel)
    {
        await Scenario(channel)
            .When().PlaceOrder().WithCountry("DE")
            .Then().ShouldSucceed()
            .And().Order()
            .HasSubtotalPrice(20.0m)
            .HasTaxRate(0.19m)
            .HasTotalPrice(23.8m);
    }

    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task OrderTotalShouldReflectCouponDiscount(Channel channel)
    {
        await Scenario(channel)
            .Given().Coupon().WithCode("DISC10").WithDiscountRate(0.1m)
            .When().PlaceOrder().WithCouponCode("DISC10")
            .Then().ShouldSucceed()
            .And().Order()
            .HasSubtotalPrice(18.0m)
            .HasDiscountRate(0.1m)
            .HasAppliedCouponCode("DISC10")
            .HasTotalPrice(18.0m);
    }

    [Theory]
    [ChannelData(ChannelType.API)]
    public async Task OrderTotalShouldApplyCouponDiscountAndTax(Channel channel)
    {
        await Scenario(channel)
            .Given().Coupon().WithCode("COMBO10").WithDiscountRate(0.1m)
            .When().PlaceOrder().WithCountry("GB").WithCouponCode("COMBO10")
            .Then().ShouldSucceed()
            .And().Order()
            .HasSubtotalPrice(18.0m)
            .HasDiscountRate(0.1m)
            .HasTaxRate(0.2m)
            .HasAppliedCouponCode("COMBO10")
            .HasTotalPrice(21.6m);
    }
}
